Include/internal/pycore_gc.h

```diff
@@ -120,6 +120,10 @@ static inline void _PyGC_SET_FINALIZED(PyObject *op) {
     PyGC_Head *gc = _Py_AS_GC(op);
     _PyGCHead_SET_FINALIZED(gc);
 }
+static inline void _PyGC_CLEAR_FINALIZED(PyObject *op) {
+    PyGC_Head *gc = _Py_AS_GC(op);
+    gc->_gc_prev &= ~_PyGC_PREV_MASK_FINALIZED;
+}
 
```

Comment:
This is expedient, but why not follow the patter used by `FINALIZED` and `SET_FINALIZED` that have a helper that skips the `_Py_AS_GC()` call?

Reply:
I think those functions (`_PyGCHead_SET_FINALIZED` and `_PyGCHead_FINALIZED`) are likely to be removed in other PRs as I work on making the GC thread-safe in free-threaded builds. It's easier to support both builds when the functions take `PyObject*` instead of `PyGC_Head*` because the free-threaded builds will store the GC flags on the `PyObject*` itself (and eventually may not even have a `PyGC_Head` prefix.

Reply:
I'm also OK with following the existing pattern for now and dealing with any potential removal later if you prefer.

Mostly I'm stuck on the warning generated in `test_async_gen_exception_12`. I don't know how to deal with it. Maybe @kumaraditya303 has suggestions?